frontend/src/metabase/visualizations/visualizations/Funnel.jsx

```diff
@@ -110,36 +111,49 @@ export default class Funnel extends Component {
       showColumnSetting: true,
       marginBottom: "0.625rem",
     }),
+    "funnel.order_dimension": {
+      getValue: (_series, settings) => settings["funnel.dimension"],
+      readDependencies: ["funnel.rows"],
+    },
     "funnel.rows": {
       section: t`Data`,
       widget: ChartSettingOrderedSimple,
-      isValid: (series, settings) => {
-        const funnelRows = settings["funnel.rows"];
 
-        if (!funnelRows || !_.isArray(funnelRows)) {
-          return false;
-        }
-        if (!funnelRows.every(setting => setting.originalIndex !== undefined)) {
-          return false;
+      getValue: (series, settings) => {
+        const seriesOrder = settings["funnel.rows"];
+        const seriesKeys = series.map(s => keyForSingleSeries(s));
+        const orderDimension = settings["funnel.order_dimension"];
+        const dimension = settings["funnel.dimension"];
+
+        const getDefault = keys =>
+          keys.map(key => ({
+            key,
+            name: key,
+            enabled: true,
+          }));
+        if (
+          !seriesOrder ||
+          !_.isArray(seriesOrder) ||
+          !seriesOrder.every(setting => setting.key !== undefined) ||
```

Comment:
nit: This checks a little differently than we do in `graph.js`
https://github.com/metabase/metabase/blob/2d170b22f3d6114076b22f73430a87df30b284b3/frontend/src/metabase/visualizations/lib/settings/graph.js#L193

We should either use `!!value` or `value !== undefined` in this case just to keep it consistent.